### PR TITLE
reports: Add Vaibhav Rai GSoC report

### PR DESCRIPTION
**Pull request by RaiVaibhav**
GSoC-2018 report

https://deploy-preview-611--coala-projects.netlify.com/gsoc/2018/statusreport/vaibhavrai

### CODE REVIEW

File: reports/vaibhav.md [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+same trace feature, so it was not possible to run coverage and debugger in
+parallel or say it is not possible to chain the trace functions, so Mischa gave
+a idea to restore the trace callbacks and then run test and finally it worked
+so I included both ideal and `Mock` test so that the coverage remain 100, but
+in all of these I lagged 3-4 week from my schedule. So I couldn't implement one
+of the proposed feature.
+
+#### Work to be done
+
+* A tool which inspect the result created by Bear.
```
[the remark this answers is not in the document]
Comment (RaiVaibhav) [reply reports/vaibhav.md]: waiting for @Makman2  response I have messaged him on gitter.

[the remark this answers is not in the document]
Comment (RaiVaibhav) [reply reports/vaibhav.md]: After that I will update the PR. (message regarding new issue for `result inspect`)

[the remark this answers is not in the document]
Comment (RaiVaibhav) [reply reports/vaibhav.md]: I have added in the last line on which task I will work after GSoC.

File: reports/vaibhav.md [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+ - GitLab:
+   - username: RaiVaibhav
+   - link: https://gitlab.com/RaiVaibhav
+ - Gitter:
+   - username: RaiVaibhav
+   - link: https://gitter.im/RaiVaibhav
+email: raivaibhav08@gmail.com
+blog: https://raivaibhav.github.io
+activity:
+ - 0:
```
Comment (jayvdb) [inline reports/vaibhav.md]: start with 1

add OpenHub to social

File: reports/vaibhav.md [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+`origin`, `diffs`, `file`, `severity`, `message`, `aspects` and several other
+attributes in debugging environment or in a seperate python console where all
+where user can easily access there values. Information about result will help
+bear developer in creating a better performant Bears.
+
+Result inspection tool will not only help in debugging the bear results but
+also help in testing the bear by asserting the various attributes of expected
+result with actual result.
+
+So, after `GSoC` my first task will be to add the inspection tool in the coala
```
Comment (jayvdb) [inline reports/vaibhav.md]: no ... you have zero responsibility after GSoC.

it is up to you if you do this or not.

if you dont, someone else will.  You've created the framework.

Comment (RaiVaibhav) [reply reports/vaibhav.md]: So I will have to add a issue link here ?

**Comment by jayvdb** [comment]
You missed a commit in https://gitlab.com/coala/mobans/graphs/master

https://gitlab.com/coala/mobans/commit/b2d62db0e7bd8879843879974f2168cc3600bfa2

File: reports/vaibhav.md [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+ - Gitter:
+   - username: RaiVaibhav
+   - link: https://gitter.im/RaiVaibhav
+ - Openhub:
+   - username: RaiVaibhav
+   - link: https://www.openhub.net/accounts/RaiVaibhav
+email: raivaibhav08@gmail.com
+blog: https://raivaibhav.github.io
+activity:
+ - 0:
```
Comment (jayvdb) [inline reports/vaibhav.md]: start numbering at 1 , not 0

https://github.com/coala/projects/issues/615

File: reports/vaibhav.md [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+project: Debug and Profile Bears
+project_link: https://summerofcode.withgoogle.com/projects/#6109762077327360
+tarball: https://rawgit.com/RaiVaibhav/GSoC-Report/master/project.tar.gz
+sha256sum : d61862f8f8be5ec168b07741c09781ef2dcc991e44d87edc05e6d237044052b0
+mentors: >
+ [Mischa Krüger](https://github.com/Makman2),
+ [Fabian Neuschmidt](https://github.com/fneu),
+ [Arjun Singh Yadav](https://github.com/arjunsinghy96)
+phase:
+ - Bonding : https://gitlab.com/coala/GSoC/GSoC-2018/milestones/18
```
Comment (jayvdb) [inline reports/vaibhav.md]: it needs to pass yamllint #615

File: reports/vaibhav.md [hunk 6]
```diff
@@ -0,0 +1,192 @@
...
+   - repo: coala
+   - link: https://github.com/coala/coala/commit/0e075217ea1bc5732d9deac57355431db00ef4ee
+   - details: >
+      Profile_Bears.rst: Add Profiling Interface document with demo.
+ - 14:
+   - repo: pytest-plugins
+   - link: https://github.com/manahl/pytest-plugins/pull/89/commits/6f8d598b210da0f76e09ff86811d4538a1bc6c43
+   - details: >
+      pytest_profiling.py: Add additional argument to configurable profile
+      output directory '(Not merged; approved by GSoC admin)'
```
Comment (jayvdb) [inline reports/vaibhav.md]: remove quotes around brackets

File: reports/vaibhav.md [hunk 7]
```diff
@@ -0,0 +1,192 @@
...
+permalink: /GSoC/2018/StatusReport/VaibhavRai
+student: Vaibhav Kumar Rai
+organisation: coala
+organisation_link: https://coala.io
+project: Debug and Profile Bears
+project_link: https://summerofcode.withgoogle.com/projects/#6109762077327360
+tarball: https://rawgit.com/RaiVaibhav/GSoC-Report/master/project.tar.gz
+sha256sum: d61862f8f8be5ec168b07741c09781ef2dcc991e44d87edc05e6d237044052b0
+mentors: >
+ [Mischa Krüger](https://github.com/Makman2),
```
Comment (jayvdb) [inline reports/vaibhav.md]: not indented correctly.

File: _reports/vaibhav.md [hunk 8]
```diff
@@ -0,0 +1,195 @@
+---
+categories:
+  - GSoC
+  - 2018
+  - StatusReport
+date: 2018-08-12
+report_year: '2018'
```
[the remark this answers is not in the document]
Comment (RJ722) [reply _reports/vaibhav.md]: I think it is still required.

[the remark this answers is not in the document]
Comment (Vamshi99) [reply _reports/vaibhav.md]: No, I don't think so. It was just for permalink.

Comment (sks444) [reply _reports/vaibhav.md]: Shit..sorry..I saw merged files has it..so i deleted my comment..:(
>You could remove it now?

File: _reports/vaibhav.md [hunk 9]
```diff
@@ -0,0 +1,195 @@
...
+  [Fabian Neuschmidt](https://github.com/fneu),
+  [Arjun Singh Yadav](https://github.com/arjunsinghy96)
+phase:
+ - Bonding: https://gitlab.com/coala/GSoC/GSoC-2018/milestones/18
+ - Phase 1: https://gitlab.com/coala/GSoC/GSoC-2018/milestones/52
+ - Phase 2: https://gitlab.com/coala/GSoC/GSoC-2018/milestones/53
+ - Phase 3: https://gitlab.com/coala/GSoC/GSoC-2018/milestones/54
+bio: >
+ I am third year student of B.Tech Computer Science Engineering at Shri Mata
+ Vaishno Devi Uiversity, Katra, Jammu and Kashmir. I participated in GSoC and
```
Comment (Vamshi99) [inline _reports/vaibhav.md]: `Uiversity` --> `University`

**Comment by RaiVaibhav** [comment]
sorry I removed the log and aciddently pushed without any change so it gets closed

File: _reports/RaiVaibhav.md [hunk 10]
```diff
@@ -0,0 +1,195 @@
...
+project: Debug and Profile Bears
+project_link: https://summerofcode.withgoogle.com/projects/#6109762077327360
+tarball: https://rawgit.com/RaiVaibhav/GSoC-Report/master/project.tar.gz
+sha256sum: d61862f8f8be5ec168b07741c09781ef2dcc991e44d87edc05e6d237044052b0
+mentors: >
+  [Mischa Krüger](https://github.com/Makman2),
+  [Fabian Neuschmidt](https://github.com/fneu),
+  [Arjun Singh Yadav](https://github.com/arjunsinghy96)
+phase:
+ - Bonding: https://gitlab.com/coala/GSoC/GSoC-2018/milestones/18
```
Comment (jayvdb) [inline _reports/RaiVaibhav.md]: https://github.com/coala/projects/pull/611#discussion_r209443765

the whole yaml needs to pass yamllint

see changes in the example

**Comment by jayvdb** [comment]
travis is failing.

**Comment by jayvdb** [comment]
ack https://github.com/coala/projects/pull/611/commits/69d97ccf9eee9b930ded3185a1677b2fff98c55e

**Comment by jayvdb** [comment]
@gitmate-bot ff

**Pull request merged**